### DISCUSSION TITLE
Could not change admin password

### DISCUSSION THREAD

**Discussion by dbarzin** [monarc-project/MonarcAppFO#497] [Q&A]
This bug tracker is for tracking bugs, todos and confirmed new features.
If you want to submit a feature request please discuss about it here:
https://github.com/monarc-project/MonarcAppFO/discussions

**Describe the bug**
Got an error the trying to change admin password

**To Reproduce**
1. Login with admin in Monarc
2. Go to Account
3. Enter previous password and new password.
4. click "Update password"

**Expected behavior**
Password changed with no error

**Screenshots**

**Desktop (please complete the following information):**

| Questions                                | Answers
|-----------------------------------------|--------------------
| Type of issue                           | Bug
| OS version (server)                 | Ubuntu 22.04
| OS version (client)                   | Win10
| PHP version                             | 8.1
| MONARC version / git hash     | 2.12.6
| Browser                                    | Chromium

**Additional context**
{
    "errors": [
        {
            "message": "Laminas\\View\\Renderer\\PhpRenderer::render: Unable to render template \"monarc/front-office/api-user-password/get\"; resolver could not resolve to a file",
            "error": "error-exception",
            "exception": {
                "class": "Laminas\\View\\Exception\\RuntimeException",
                "file": "/var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php",
                "line": 518,
                "message": "Laminas\\View\\Renderer\\PhpRenderer::render: Unable to render template \"monarc/front-office/api-user-password/get\"; resolver could not resolve to a file",
                "stacktrace": "#0 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-view/src/View.php(194): Laminas\\View\\Renderer\\PhpRenderer->render()\n#1 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-view/src/View.php(222): Laminas\\View\\View->render()\n#2 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-view/src/View.php(187): Laminas\\View\\View->renderChildren()\n#3 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/View/Http/DefaultRenderingStrategy.php(92): Laminas\\View\\View->render()\n#4 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(320): Laminas\\Mvc\\View\\Http\\DefaultRenderingStrategy->render()\n#5 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(170): Laminas\\EventManager\\EventManager->triggerListeners()\n#6 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Application.php(354): Laminas\\EventManager\\EventManager->triggerEvent()\n#7 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Application.php(335): Laminas\\Mvc\\Application->completeRequest()\n#8 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/public/index.php(34): Laminas\\Mvc\\Application->run()\n#9 {main}"
            }
        }
    ]
}

**Comment by ruslanbaidan** [comment]
Thank you for the report.
It seems the error message that you receive is due to some rendering problems, but there is an issue with the password change due to a warning message. We have some legacy parts in the code that are going to be improved.

php v8.x comes with turned on warnings and notices, so could you try to tun them off in your php.ini file:
`error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT & ~E_NOTICE & ~E_WARNING`
and reload apache server.

**Comment by dbarzin** [comment]
Shouldn't it be added in the installation procedure ?

**Comment by dbarzin** [comment]
After this change, I got another error : 

angular-common-libs.js:132  Possibly unhandled rejection: {"data":{"errors":[{"message":"Monarc\\Core\\Service\\PasswordService::changePassword(): Argument #3 ($newPassword) must be of type string, null given, called in /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/monarc/frontoffice/src/Controller/ApiUserPasswordController.php on line 37","error":"error-exception","exception":{"class":"TypeError","file":"/var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/monarc/core/src/Service/PasswordService.php","line":123,"message":"Monarc\\Core\\Service\\PasswordService::changePassword(): Argument #3 ($newPassword) must be of type string, null given, called in /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/monarc/frontoffice/src/Controller/ApiUserPasswordController.php on line 37","stacktrace":"#0 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/monarc/frontoffice/src/Controller/ApiUserPasswordController.php(37): Monarc\\Core\\Service\\PasswordService->changePassword()\n#1 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Controller/AbstractRestfulController.php(431): Monarc\\FrontOffice\\Controller\\ApiUserPasswordController->update()\n#2 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(320): Laminas\\Mvc\\Controller\\AbstractRestfulController->onDispatch()\n#3 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(178): Laminas\\EventManager\\EventManager->triggerListeners()\n#4 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Controller/AbstractController.php(105): Laminas\\EventManager\\EventManager->triggerEventUntil()\n#5 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Controller/AbstractRestfulController.php(306): Laminas\\Mvc\\Controller\\AbstractController->dispatch()\n#6 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/DispatchListener.php(117): Laminas\\Mvc\\Controller\\AbstractRestfulController->dispatch()\n#7 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(320): Laminas\\Mvc\\DispatchListener->onDispatch()\n#8 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-eventmanager/src/EventManager.php(178): Laminas\\EventManager\\EventManager->triggerListeners()\n#9 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/vendor/laminas/laminas-mvc/src/Application.php(319): Laminas\\EventManager\\EventManager->triggerEventUntil()\n#10 /var/lib/monarc/releases/MonarcAppFO-v2.12.6/public/index.php(34): Laminas\\Mvc\\Application->run()\n#11 {main}"}}]},"status":500,"config":{"method":"PUT","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"api/user/password/1","data":{"old":"admin"},"headers":{"Accept":"application/json, text/plain, */*","Content-Type":"application/json;charset=utf-8","token":"0f866b00663814b5e3bad76589cfb7c028c7deda85d81abe9d636cb8b5a40146479e338865014890138518.00407476"},"cached":false},"statusText":"Internal Server Error","xhrStatus":"complete"}

**Reply by ruslanbaidan** [reply]
This may happen due to a password that doesn't meet the minimum requirements of:
```
Your password is too short (min 9 characters).
Your password must contain at least: one uppercase, one number and one special character.
```
It seems the error message is not displayed on the change password page and that's why frontend sends an empty password to backend.
This will be fixed in the next release.
Could you please try to specify a password that meets the requirements.

**Comment by dbarzin** [comment]
Thanks, it works.
